woocommerce/woocommerce
# Add "Shared with you" badge to Extensions > My Subscriptions list as applicable

### Changes proposed in this Pull Request:

Fixes [WCCOM#19063](https://github.com/Automattic/woocommerce.com/issues/19063).

To make it easier to see when an extension is available because of a shared subscription, this PR amends the Extensions > My Subscriptions page for connected stores to:

- Add a "Shared with you" badge to products available as a result of a shared subscription
    - In a popup, shows the email address of the person who shared the subscription with you and a link to "learn more"
- Removes the "Manage on Woo.com" link from the actions menu for subscriptions shared with you
- Reduces the font size on such badges to 12px, in accordance with the latest designs
- Improves the spacing of these badges, where multiple badges appear on multiple lines, so they don't collide

![Screenshot highlighting the new badge and edited action menu.](https://github.com/woocommerce/woocommerce/assets/53293/24856ee4-afa6-477e-9d2b-9c77792aa295)



### How to test the changes in this Pull Request:

1. Connect a store using a Woo.com account that has at least one subscription shared with it.
2. In WooCommerce > Extensions > My Subscriptions, observe the new "Shared with you" badge.
3. Click on the badge to see the email address of the person sharing it and a link to the sharing documentation.
4. Note that the font size of the badge is now 12px.
5. Click the kebab menu menu and observe that the "Manage on Woo.com" link does not appear for shared subscriptions.
6. (Optional) Force a second badge to appear, e.g. by manually editing `plugins/woocommerce-admin/client/marketplace/components/my-subscriptions/table/rows/functions.tsx` and adding another `<StatusPopover>`; observe that the badges have both horizontal _and now vertical_ margins to prevent them from colliding with one another.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

For shared subscriptions, My Subscriptions now shows "Shared with you" and the email address of the person who shared it with you.

#### Comment 

</details>
